### PR TITLE
Add additional payments slip layouts module

### PR DESCRIPTION
**Pull request by nbessi**
And refactor payment slip to allows a more flexible print configuration system.
This should allows to set print options by printer and report in the future

### CODE REVIEW

**Comment by fclementic2c** [comment]
@nbessi could you please add the transaltion 'Invoice with payment slip' -> 'Facture avec BVR'

@lsimonin : Could tell us if you managed to merge this PR and test properly ? thanks

**Comment by lsimonin** [comment]
Thank you add the image of the Payment Slip, even if the option is not checked in the settings of the company. This will be used to send the email.

**Comment by nbessi** [comment]
@lsimonin we have an issue with background as the setting is global. If we change it will impact all bvr report. We have to think of a way to enable a more flexible setting for this option. Maybe move it on the res.partner.bank or base it on report name, or using a condition as in email template. 

I ll come back to you when I have a solution.

**Comment by lsimonin** [comment]
@nbessi 
You can add two check box in the Data tab of the BVR company
- Activate the report Invoice + Payment Slip with background image
- Activate the report Invoice + Payment Slip without background image

As we have merged two reports with 2 options.

**Comment by nbessi** [comment]
@lsimonin the functionality should be available

**Comment by lsimonin** [comment]
@nbessi it work's fine.Everything is ok for me, you can merge

**Comment by lsimonin** [comment]
:+1: ;)

File: l10n_ch_payment_slip/report/ir_action.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                cr,
+                name
+            )
+
+    def render_report(self, cr, uid, res_ids, name, data, context=None):
+        """
+        Override to work with real pdf in testing and not html
+        as we use reportlab as renderer
+        """
+        if context.get('force_pdf'):
```
Comment (yvaucher) [inline l10n_ch_payment_slip/report/ir_action.py:57]: You should probably ensure context is not None

File: l10n_ch_payment_slip_layouts/tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# -*- coding: utf-8 -*-
```
Comment (yvaucher) [inline l10n_ch_payment_slip_layouts/tests/__init__.py]: Twice there

File: l10n_ch_payment_slip_layouts/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{'name': 'Payment slip alternate layout(s)',
+ 'version': '0.1',
```
Comment (yvaucher) [inline l10n_ch_payment_slip_layouts/__openerp__.py:22]: Can you add a summary ?

**Comment by lsimonin** [comment]
@nbessi Can you merge?

**Comment by yvaucher** [comment]
@lsimonin I raised few things to fix first

**Comment by nbessi** [comment]
@yvaucher fixed pushed waiting for travis

File: l10n_ch_payment_slip_layouts/__openerp__.py (hunk 3 above)
Comment (nbessi) [inline l10n_ch_payment_slip_layouts/__openerp__.py:22]: Done

**Comment by nbessi** [comment]
@yvaucher travis is green

**Comment by yvaucher** [comment]
:+1: Thanks for the fixes

**Comment by lsimonin** [comment]
:+1:

**Pull request merged**